File: dane-do-planu/Dane do planu/group.py
import requests
import json
import mysql.connector
import string

def fetch_group_data(group_number):
    url = f"https://plan.zut.edu.pl/schedule.php?kind=group&query={group_number}"
    headers = {
        "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36"
    }
    response = requests.get(url, headers=headers)

    if response.status_code != 200:
        print(f"Error fetching data for group {group_number}: HTTP {response.status_code}")
        return None

    try:
        data = json.loads(response.text)
        group_names = [item["item"] for item in data if isinstance(item, dict) and "item" in item]
        return [(name, 0) for name in group_names]
    except json.JSONDecodeError as e:
        print(f"Error decoding JSON for group {group_number}: {e}")
        return None

def insert_into_database(group_data):
    try:
        connection = mysql.connector.connect(
            host='localhost',
            user='root',
            password='B7uWyeGcjqRX3bv!',
            database='lepszy_plan'
        )
        cursor = connection.cursor()

        insert_query = """
        INSERT INTO `Group` (group_name)
        VALUES (%s)
        """
        cursor.executemany(insert_query, [(name,) for name, _ in group_data])
        connection.commit()

        print(f"Inserted {len(group_data)} groups into the database.")
    except mysql.connector.Error as err:
        print(f"Database error: {err}")
    finally:
        if connection.is_connected():
            cursor.close()
            connection.close()

if __name__ == "__main__":
    for group_number in string.ascii_uppercase:
        group_data = fetch_group_data(group_number)
        if group_data:
            insert_into_database(group_data)

File: dane-do-planu/Dane do planu/Group.php
<?php

function fetchGroupData($groupNumber) {
    $url = "https://plan.zut.edu.pl/schedule.php?kind=group&query=" . urlencode($groupNumber);
    
    $options = [
        "http" => [
            "header" => "User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36\r\n",
            "ignore_errors" => true // this ensures the content is returned even if HTTP code is not 200
        ],
        "ssl" => [
            "verify_peer" => false, // Disable SSL verification
            "verify_peer_name" => false, // Disable peer name verification
        ]
    ];

    $context = stream_context_create($options);
    $response = file_get_contents($url, false, $context);

    if ($response === false) {
        echo "Error fetching data for group $groupNumber\n";
        return null;
    }

    $data = json_decode($response, true);

    if (json_last_error() !== JSON_ERROR_NONE) {
        echo "Error decoding JSON for group $groupNumber: " . json_last_error_msg() . "\n";
        return null;
    }

    $groupNames = [];

    foreach ($data as $item) {
        if (is_array($item) && isset($item['item'])) {
            $groupNames[] = [$item['item'], 0];
        }
    }

    return $groupNames;
}

function insertIntoDatabase($groupData) {
    try {
        $pdo = new PDO('sqlite:database.db');
        $pdo->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);

        // Assuming you want to use SQLite, leave it as INSERT OR IGNORE
        $insertQuery = "INSERT OR IGNORE INTO `Group` (group_name) VALUES (:group_name)";
        $stmt = $pdo->prepare($insertQuery);

        foreach ($groupData as $group) {
            $stmt->execute(['group_name' => $group[0]]);
        }

        echo "Inserted " . count($groupData) . " groups into the database.\n";
    } catch (PDOException $e) {
        echo "Database error: " . $e->getMessage() . "\n";
    }
}

for ($groupNumber = 'A'; $groupNumber <= 'Z'; $groupNumber++) {
    $groupData = fetchGroupData($groupNumber);
    if ($groupData) {
        insertIntoDatabase($groupData);
    }
}
echo "Process completed - Group.\n";